templates/registry/registry-dpl.yaml

```diff
@@ -36,14 +36,56 @@ spec:
 {{- end }}
     spec:
       securityContext:
-        fsGroup: 10000
+        fsGroup: 10001
 {{- if .Values.registry.serviceAccountName }}
       serviceAccountName: {{ .Values.registry.serviceAccountName }}
 {{- end -}}
       {{- with .Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      initContainers:
+      - name: "change-pemission-of-directory"
```

Comment:
You don't need this init container, because the securityContext.fsGroup is already set to 10001 (registry group) and the entrypoint already uses umask 0002 (which takes care of the chmod part).

Reply:
Well I got rid of the entrypoint...

Reply:
This is one thing we can't get rid of in the entrypoint, because without setting `umask 0002`, new files/folders are not created as write-able by the `registry` group, which will lead to garbage collection failures (short story: files created by docker-registry in the `registry` container cannot be removed by Harbor in the `registryctl` container).
I already checked, there isn't a UMASK env variable that we could use as a Dockerfile option (that would've been nicer).